wgpu-core/src/validation.rs

```diff
@@ -1456,12 +1480,47 @@ impl Interface {
         }
 
         match shader_stage {
-            naga::ShaderStage::Vertex => {
+            ShaderStageForValidation::Vertex {
+                topology,
+                compare_function,
+            } => {
+                let mut max_vertex_shader_output_variables =
+                    self.limits.max_inter_stage_shader_variables;
+                let mut max_vertex_shader_output_location = max_vertex_shader_output_variables - 1;
+
+                let point_list_deduction = if topology == wgt::PrimitiveTopology::PointList {
+                    Some(MaxVertexShaderOutputDeduction::PointListPrimitiveTopology)
+                } else {
+                    None
+                };
+
+                let deductions = point_list_deduction.into_iter();
+
+                for deduction in deductions.clone() {
+                    max_vertex_shader_output_variables = max_vertex_shader_output_variables
+                        .checked_sub(deduction.for_variables())
+                        .unwrap();
+                    max_vertex_shader_output_location = max_vertex_shader_output_location
+                        .checked_sub(deduction.for_location())
+                        .unwrap();
```

Comment:
It's not that much more work to raise an error here. How sure can we be that this (or more realistically, the fragment shader one) will never underflow? (Fine to wait for a follow-up, even if it can.)